01-import_and_filter.py

```diff
@@ -3,32 +3,65 @@
 01. Filter using MNE-python
 ===========================
 
-XXX: this needs update so that it references config.h_freq config.l_blablabla
-
-The data are bandpass filtered (1 - 40 Hz) using linear-phase fir filter with
-delay compensation. For the lowpass filter the transition bandwidth is
-automatically defined. See
-`Background information on filtering <http://mne-tools.github.io/dev/auto_tutorials/plot_background_filtering.html>`_
+The data are bandpass filtered to the frequencies defined in config.py
+(config.h_freq - config.l_freq Hz) using linear-phase fir filter with
+delay compensation.
+The transition bandwidth is automatically defined. See
+`Background information on filtering
+<http://mne-tools.github.io/dev/auto_tutorials/plot_background_filtering.html>`_
 for more. The filtered data are saved to separate files to the subject's'MEG'
 directory.
+If config.plot = True plots raw data and power spectral density.
 """  # noqa: E501
 
 import os.path as op
 
 import mne
 from mne.parallel import parallel_func
-
+import glob
 import config
 
+
 def run_filter(subject):
     print("processing subject: %s" % subject)
-    # XXX : put the study-specific names in the config file
+
     meg_subject_dir = op.join(config.meg_dir, subject)
-    raw_fnames_in = [op.join(meg_subject_dir, '%s_audvis_raw.fif' % subject)]
-    raw_fnames_out = [op.join(meg_subject_dir, '%s_audvis_filt_raw.fif' % subject)]
 
+    # try to find multiple runs
+    raw_fnames_in = sorted(glob.glob(op.join(meg_subject_dir,
+                                             '%s_%s_run??_raw.fif'
+                                             % (subject, config.study_name))))
```

Comment:
maybe this is dangerous as if you run twice the scripts, the second time you'll have more files that end with raw.fif

How about using modern python string formatting:

```
In [1]: subject = 'sub01'
In [2]: run = 'run1'
In [3]: '{subject}_{run}_raw.fif'.format(run=run, subject=subject)
Out[3]: 'sub01_run1_raw.fif'
```

basically the format `'{subject}_run{run}_raw.fif'` would be part of the config file
as well as the list of runs.

Something like this:


```
subject = 'sub01'

runs = ['run1', 'run2', 'run3']
base_raw_fname = '{subject}_{run}_raw.fif'

# then every script does:

for run in runs:
    raw_fname = base_raw_fname.format(**locals())
    print(raw_fname)
```

basically I find the use of glob too dangerous

Reply:
I see. 
shall we make 
``
runs = ['run1', 'run2', 'run3']
``
a dict to allow for different number of runs per participant (real world users problems)